athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java

```diff
@@ -795,13 +795,13 @@ else if (value instanceof Long) {
                 case VARBINARY:
                     if (value instanceof ArrowBuf) {
                         ArrowBuf buf = (ArrowBuf) value;
-                        writer.varBinary().writeVarBinary(0, buf.capacity(), buf);
+                        writer.varBinary().writeVarBinary(0, (int) (buf.capacity()), buf);
                     }
                     else if (value instanceof byte[]) {
                         byte[] bytes = (byte[]) value;
                         try (ArrowBuf buf = allocator.buffer(bytes.length)) {
                             buf.writeBytes(bytes);
-                            writer.varBinary().writeVarBinary(0, buf.readableBytes(), buf);
+                            writer.varBinary().writeVarBinary(0, (int) (buf.readableBytes()), buf);
```

Comment:
What type did they change `readableBytes` and `capacity` to?

Reply:
they changed both to long. I took a look at the underlying implementation, and it didn't seem that theres a chance for overflow, but i'm not sure if theres any good way to do this since writeVarBinary expects an int.